### PR TITLE
Upgrade third-party llvm to llvm-12

### PR DESCRIPTION
**Pull request by daviditen**
Update our bundled LLVM to LLVM-12. Update a build script to allow either 12 or 11.

Update some of our usage of LLVM in the compiler to fix some breaking changes between versions.

Update a test whose debug output changed slightly between LLVM versions.

This has been running in the performance playground and hasn't found any significant performance differences.

### CODE REVIEW

**Comment by daviditen** [comment]
Passed linux64 paratest with CHPL_LLVM=bundled.

**Comment by ronawho** [comment]
Just as a meta note, I think we should wait until after the point release to merge this unless you see a strong reason to try and get it in.

File: third-party/llvm/README
Comment (ronawho) [inline third-party/llvm/README:5]: Note the "how to upgrade" instructions at the end need to be updated

**Comment by daviditen** [comment]
@ronawho I updated the version numbers in the documentation. Does it look good to merge after the point release?

**Review by ronawho — APPROVED** [review]

**Comment by daviditen** [comment]
Thanks!

**Comment by bradcray** [comment]
@daviditen: The QUICKSTART.rst file also indicates which LLVM version(s) are supported (the rationale being so that someone truly impatient who's only reading the quickstart will not need to read anything more than that).  Would you update it as well?

**Comment by daviditen** [comment]
https://github.com/chapel-lang/chapel/pull/19183 updates QUICKSTART.rst.

**Pull request merged**
